sunxspex/integrate.py

```diff
@@ -0,0 +1,192 @@
+import numpy as np
+from scipy.integrate._quadrature import _cached_roots_legendre
+from scipy.special import lpmv
+
```

Comment:
could consider using [autoray](https://github.com/jcmgray/autoray) for true freedom of backend. Probably only useful if the fitting infrastructure uses it too, so that tensorflow, torch, jax etc. could be used to speed that up if you've got a GPU on hand.